specs/devexperience/DevExperience.md

```diff
@@ -0,0 +1,181 @@
+# Reunion Developer Experience
+
+## 1.0 - Overview
+This spec will detail the design of how the Reunion developer experience will work. Through analysis of the current landscape of Windows development, there are a few important items that were found:
+1.	The Windows Application Packaging (WAP) project increases build time by 40%
+2.	Requiring the WAP creates a disjointed developer experience, where developers trying to create Win32 apps have a different workflow than UWP developers.
+3.	Therefore, to provide our customers with the best inner-loop, and to unify the experience for Win32 and UWP developers, we must not require the WAP for our core user scenarios.
+
+If we are to remove the WAP project, we then must ask ourselves the following questions:
+1.	What kicks in the MSIX Package tooling?
+2.	Can developers opt-out of the pri generation?
+3.	How does building a Sparse Package look?
+    - A Sparse Package is a Win32 app that has identity, but isn’t using MSIX.
+    - This means there is a package.appxmanifest and a SxS manifest
+4.	What to do with the Windows store /start menu assets that aren’t used in unpackaged applications
+5.	How does an existing UWP developer move forward and use this?
+6.	How does an existing WPF or WinForms developer move forward and use this?
+
+## 2.0 – Definitions
+- Reunion – The effort to unify the Win32 and UWP platforms
+- Reunion Tooling – MSBuild and VisualStudio tooling used to build Reunion apps.
+- Workflow – The steps a developer will take in Visual Studio to be successful using Reunion.
+- WAP – A project used to generate an .msix or .msixbundle.
+- Sparse Package – An unpackaged Win32 app with package Identity via registration of an .msix.
+
+## 3.0 Guiding Principles
+Below is a list of some guiding principles to ground this proposal
+1.	Easy for developers to understand.
+2.	Easy for developers to change between the four different app model types:
+	- Unpackaged
+    - Sparse Package
```

Comment:
These "app model types" seem a weird way of specifying what a developer wants to achieve.
eg 
The app needs to be able to be xcopy deployed = Unpackaged
The app needs to be able to be xcopy deployed but wants to leverage identity = Sparse Package
The app needs to run in the Windows App Container = Packaged Desktop with Trust Level = Partial <-- is this correct?
The app needs to have a modern app lifecycle = UWP packaging <-- Seriously don't call this Universal or UWP. 

The names of the app model types should reflect the impact they have on the app, rather than some historic name that's not fully understood by anyone.